src/main/java/com/ericsson/eiffelcommons/utils/HttpRequest.java

```diff
@@ -264,6 +267,23 @@ public HttpRequest setBody(File file, ContentType contentType) throws IOExceptio
         return setBody(fileContent, contentType);
     }
 
+    /**
+     * Function that sets the Authorization header of the http request.
+     *
+     * @param httpRequest
+     * @param username
+     * @param password
+     * @return
+     * @throws UnsupportedEncodingException
+     */
+    public HttpRequest setCredentials(HttpRequest httpRequest, String username,
```

Comment:
should we have a setter method for adding credentials, can this be added to the constructor instead?

Reply:
Not all request will require credentials. A setter for token may also be needed.

Reply:
the credentials can be provided as an optional constructor, if it has uname and password it will set them. 
may be different constructors for token, uname/password and without any credentials.
It is not good to ask username and password. instead we can ask for Credentials that was set with uname and password.

Reply:
I think it's better to use it as christoffer has done. It way more flexible and the class isnt cluttered with different constructors. Now you create your request object and then you manipulate it to your will 👿 😃

Reply:
There are already 4 constructors in this class. If we add username and password as an option then I don't know which combination of parameters should be supported with these added parameters. We might be looking at 8 constructors then.